src/audio/detect_test.c

```diff
@@ -698,15 +698,15 @@ static int test_keyword_trigger(struct comp_dev *dev, int cmd)
 
 	ret = comp_set_state(dev, cmd);
 	if (ret)
-		return ret;
+		goto err;
 
 	if (cmd == COMP_TRIGGER_START ||
 	    cmd == COMP_TRIGGER_RELEASE) {
 		cd->detect_preamble = 0;
 		cd->detected = 0;
 		cd->activation = 0;
 	}
-
+err:
```

Comment:
an alternative would be `return 0;` below

Reply:
yes, done